### ISSUE TITLE
HLS does not report errors

### ISSUE THREAD

**Issue by asivitz** [haskell/haskell-language-server#3840]
### Your environment

Which OS do you use?
MacOS
Which version of GHC do you use and how did you install it?
9.4.5 from ghcup
How is your project built (alternative: link to the project)?
stack

Which LSP client (editor/plugin) do you use?
Neovim+LanguageClient-neovim
Which version of HLS do you use and how did you install it?
2.2.0.0 from ghcup
Have you configured HLS in any way (especially: a `hie.yaml` file)?
Yes, a hie.yaml file
### Steps to reproduce
haskell-language-server-wrapper --debug core/Hickory/Vulkan/Mesh.hs
### Expected behaviour
HLS should report an error that I've introduced
### Actual behaviour
HLS reports only warnings
### Debug information

Everything had been working fine for months. I changed my GHC version for an unrelated reason, and now I can't get HLS working again, even after reverting my GHC version. I've tried making sure the project builds before starting HLS. I've tried blowing away my hie-bios cache. I've tried changing the cradle to cabal. The --debug log does not indicate any errors.

Here is the `haskell-language-server-wrapper --debug` output when I run it on a file with an error that I've introduced:

```
Found "/Users/user/dev/Hickory/hie.yaml" for "/Users/user/dev/Hickory/a"
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 2.2.0.0 aarch64 ghc-9.4.5
Current directory: /Users/user/dev/Hickory
Operating system: darwin
Arguments: ["--debug","core/Hickory/Vulkan/Mesh.hs"]
Cradle directory: /Users/user/dev/Hickory
Cradle type: Stack

Tool versions found on the $PATH
cabal:          3.6.2.0
stack:          2.11.1
ghc:            9.4.5


Consulting the cradle to get project GHC version...
2023-10-11T19:34:33.106560Z | Debug | executing command: stack setup --silent
2023-10-11T19:34:33.136258Z | Debug | executing command: stack exec ghc -- --numeric-version
Project GHC version: 9.4.5
haskell-language-server exe candidates: ["haskell-language-server-9.4.5","haskell-language-server"]
Launching haskell-language-server exe at:/Users/user/.ghcup/bin/haskell-language-server-9.4.5
2023-10-11T19:34:33.396800Z | Debug | executing command: stack setup --silent
2023-10-11T19:34:33.416111Z | Debug | executing command: stack exec ghc -- -v0 -package-env=- -ignore-dot-ghci -e Control.Monad.join (Control.Monad.fmap System.IO.putStr System.Environment.getExecutablePath)
2023-10-11T19:34:33.844794Z | Debug | executing command: stack setup --silent
2023-10-11T19:34:33.861879Z | Debug | executing command: stack exec ghc -- --print-libdir
2023-10-11T19:34:34.231889Z | Info | haskell-language-server version: 2.2.0.0 (GHC: 9.4.5) (PATH: /Users/user/.ghcup/bin/haskell-language-server-9.4.5~2.2.0.0)
2023-10-11T19:34:34.232635Z | Info | Directory: /Users/user/dev/Hickory
2023-10-11T19:34:34.232885Z | Info | Logging heap statistics every 60.00s
 ghcide setup tester in /Users/user/dev/Hickory.
Report bugs at https://github.com/haskell/haskell-language-server/issues

Step 1/4: Finding files to test in /Users/user/dev/Hickory
Found 1 files

Step 2/4: Looking for hie.yaml files that control setup
Found 1 cradle
  (/Users/user/dev/Hickory/hie.yaml)

Step 3/4: Initializing the IDE
2023-10-11T19:34:34.236060Z | Debug | Initializing exports map from hiedb
2023-10-11T19:34:34.236415Z | Debug | Shake session initialized

Step 4/4: Type checking the files
2023-10-11T19:34:34.237986Z | Info | Cradle path: core/Hickory/Vulkan/Mesh.hs
2023-10-11T19:34:34.238340Z | Debug | Cradle: Cradle {cradleRootDir = "/Users/user/dev/Hickory", cradleOptsProg = CradleAction: Stack}
2023-10-11T19:34:34.238378Z | Info | invoking build tool to determine build flags (this may take some time depending on the cache)
2023-10-11T19:34:34.238509Z | Debug | executing command: stack repl --no-nix-pure --with-ghc /Users/user/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 Hickory:lib
2023-10-11T19:34:34.259573Z | Debug | Done initializing exports map from hiedb. Size: 1725
2023-10-11T19:34:34.861135Z | Debug | Building all executables for Hickory once. After a successful build of all of them, only specified
2023-10-11T19:34:34.861232Z | Debug | executables will be rebuilt.
2023-10-11T19:34:34.862678Z | Debug | Hickory> initial-build-steps (lib + exe)
2023-10-11T19:34:35.170188Z | Debug | Configuring GHCi with the following packages: Hickory.
2023-10-11T19:34:35.229703Z | Debug | executing command: stack path --ghc-package-path
2023-10-11T19:34:35.559109Z | Debug | /Users/user/dev/Hickory/.stack-work/install/aarch64-osx/40d6eaa01cf3bdefe4c74231ff72976e6f1730b3604d0e97ba2a4eba75fb9846/9.4.5/pkgdb:/Users/user/.stack/snapshots/aarch64-osx/40d6eaa01cf3bdefe4c74231ff72976e6f1730b3604d0e97ba2a4eba75fb9846/9.4.5/pkgdb:/Users/user/.ghcup/ghc/9.4.5/lib/ghc-9.4.5/lib/package.conf.d
2023-10-11T19:34:35.570722Z | Debug | executing command: stack setup --silent
2023-10-11T19:34:35.588775Z | Debug | executing command: stack exec ghc -- --print-libdir
2023-10-11T19:34:35.807813Z | Debug | Session loading result: Right (ComponentOptions {componentOptions = ["-i","-odir=/Users/user/dev/Hickory/.stack-work/odir","-hidir=/Users/user/dev/Hickory/.stack-work/odir","-hide-all-packages","-XGHC2021","-XNamedFieldPuns","-XRank2Types","-XBangPatterns","-XOverloadedStrings","-XScopedTypeVariables","-XLambdaCase","-XTupleSections","-XViewPatterns","-XBlockArguments","-XRecordWildCards","-i/Users/user/dev/Hickory/core/.stack-work/dist/aarch64-osx/Cabal-3.8.1.0/build","-i/Users/user/dev/Hickory/core","-i/Users/user/dev/Hickory/core/.stack-work/dist/aarch64-osx/Cabal-3.8.1.0/build/autogen","-i/Users/user/dev/Hickory/core/.stack-work/dist/aarch64-osx/Cabal-3.8.1.0/build/global-autogen","-stubdir=/Users/user/dev/Hickory/core/.stack-work/dist/aarch64-osx/Cabal-3.8.1.0/build","-package-id=JuicyPixels-3.3.8-LMJRHX36Q1pCfrybbsLoAQ","-package-id=JuicyPixels-extra-0.6.0-IRuyMxa4cveHupLjkuuGTY","-package-id=MonadRandom-0.6-LJRRwv297tB5cfCD4QSELn","-package-id=OpenGLRaw-3.3.4.1-Jmq4DOu5fKBApZdpiQxZQW","-package-id=VulkanMemoryAllocator-0.10.4-8O9XYClSph1Ln4AWVzoG6L","-package-id=acquire-0.2.0.1-DrrtP5mTs3fLIppLfTCpTH","-package-id=aeson-2.1.2.1-L6r5zUV5ZqfK3IRR7Jce1Q","-package-id=attoparsec-0.14.4-4wRD3aq3dD2JC9DOrhZWgM","-package-id=base-4.17.1.0","-package-id=binary-0.8.9.1","-package-id=bitwise-enum-1.0.1.0-LNZjI7M0Qyk1bEWpOKudl2","-package-id=bytestring-0.11.4.0","-package-id=containers-0.6.7","-package-id=derive-storable-0.3.1.0-HnUVgAwsIVz2N70Mf0Bmsi","-package-id=deriving-aeson-0.2.9-IerZv7M4HZREu3Fa019JZu","-package-id=directory-1.3.7.1","-package-id=fixed-vector-1.2.2.1-BOkHx28sjoSJZfJaWnDc6P","-package-id=generic-lens-2.2.2.0-ClCHvQRNkBq4DrbvZEszCe","-package-id=groom-0.1.2.1-6XAA5pv25owK2qjNUFpXHz","-package-id=hashable-1.4.2.0-GSowZAAqge792LlNNo7CcE","-package-id=lens-5.2.2-6P2dgWTizWZ2PJrZAOlDWZ","-package-id=linear-1.22-H26bHmaDVgM6RFHqQqbl3V","-package-id=managed-1.0.10-3dxDK24Vsu2AqGtDnQ3YHQ","-package-id=megaparsec-9.3.1-J6dxZOCJSbBFhU39HHitVY","-package-id=mtl-2.2.2","-package-id=random-1.2.1.1-CqDzcwYpmtdGsBnwctblrv","-package-id=rawstring-qm-0.2.3.0-1MbDKr9xFBwHzZHJfVHTZR","-package-id=reactive-banana-1.3.1.0-39MsLV9iedH8ySMZcqhwC2","-package-id=safe-0.3.19-3Ab0OpSRliCKMhc0t4Nhds","-package-id=scientific-0.3.7.0-1vOOsoPGEUcGIzDE4J1xpm","-package-id=text-2.0.2","-package-id=time-1.12.2","-package-id=unordered-containers-0.2.19.1-3vgeKBWOEKa2LoZ9GjinL2","-package-id=uuid-1.3.15-EUVyjtmQpCkAcSmwQ1J65p","-package-id=vector-0.13.0.0-JvSw347GJH4LQff8qpFgEs","-package-id=vector-binary-instances-0.2.5.2-BBxNSU6HmYR6XNwB6tJidG","-package-id=vulkan-3.23-AGoWLOV9vey94SsZ6l83FG","-package-id=vulkan-utils-0.5.10.1-7Lv4gZHKzYZ7HM4pj2kOSJ","-package-id=wavefront-0.7.1.4-Jyh2g9YKG5ILNrfzslJdsD","-Wall","-haddock","-optP-include","-optP/Users/user/dev/Hickory/.stack-work/ghci/022b4f6c/cabal_macros.h","-ghci-script=/Users/user/.cache/stack/ghci-script/d54b480c/ghci-script","-package-db","/Users/user/dev/Hickory/.stack-work/install/aarch64-osx/40d6eaa01cf3bdefe4c74231ff72976e6f1730b3604d0e97ba2a4eba75fb9846/9.4.5/pkgdb","-package-db","/Users/user/.stack/snapshots/aarch64-osx/40d6eaa01cf3bdefe4c74231ff72976e6f1730b3604d0e97ba2a4eba75fb9846/9.4.5/pkgdb","-package-db","/Users/user/.ghcup/ghc/9.4.5/lib/ghc-9.4.5/lib/package.conf.d"], componentRoot = "/Users/user/dev/Hickory/core", componentDependencies = ["core/Hickory.cabal","core/package.yaml","stack.yaml"]},"/Users/user/.ghcup/ghc/9.4.5/lib/ghc-9.4.5/lib")
2023-10-11T19:34:35.857780Z | Info | Interface files cache directory: /Users/user/.cache/ghcide/main-7ee3613dc5682aa8cbb222c7df23529ea959882c-7ee3613dc5682aa8cbb222c7df23529ea959882c
2023-10-11T19:34:35.858161Z | Info | Making new HscEnv. In-place unit ids: [ main-7ee3613dc5682aa8cbb222c7df23529ea959882c ]
2023-10-11T19:34:35.878074Z | Debug | New component cache HscEnvEq: (([],Just HscEnvEq 4),fromList [("/Users/user/dev/Hickory/hie.yaml",Just 2023-10-11 18:12:59.019730906 UTC),("core/Hickory.cabal",Just 2023-10-11 18:32:44.612010104 UTC),("core/package.yaml",Just 2023-10-11 18:31:33.921366896 UTC),("stack.yaml",Just 2023-10-11 19:26:25.942020654 UTC)])
2023-10-11T19:34:35.920489Z | Debug | Known files updated:
  fromList [(TargetModule (ModuleName "Hickory.ModelLoading.DirectX"),fromList ["/Users/user/dev/Hickory/core/Hickory/ModelLoading/DirectX.hs"]),(TargetModule (ModuleName "Hickory.Math.VectorMatrix"),fromList ["/Users/user/dev/Hickory/core/Hickory/Math/VectorMatrix.hs"]),(TargetModule (ModuleName "Hickory.Graphics.Mesh"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/Mesh.hs"]),(TargetModule (ModuleName "Hickory.ModelLoading.Animation"),fromList ["/Users/user/dev/Hickory/core/Hickory/ModelLoading/Animation.hs"]),(TargetModule (ModuleName "Hickory.Graphics.Uniforms"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/Uniforms.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Material"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Material.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Forward.ShaderDefinitions"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Forward/ShaderDefinitions.hs"]),(TargetModule (ModuleName "Hickory.Math"),fromList ["/Users/user/dev/Hickory/core/Hickory/Math.hs"]),(TargetModule (ModuleName "Hickory.Graphics.Debug"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/Debug.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Frame"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Frame.hs"]),(TargetModule (ModuleName "Hickory.Graphics.GLSupport"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/GLSupport.hs"]),(TargetModule (ModuleName "Hickory.Serialize"),fromList ["/Users/user/dev/Hickory/core/Hickory/Serialize.hs"]),(TargetModule (ModuleName "Hickory.Graphics"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Instance"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Instance.hs"]),(TargetModule (ModuleName "Hickory.Graphics.Framebuffer"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/Framebuffer.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Types"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Types.hs"]),(TargetModule (ModuleName "Hickory.Graphics.Drawing"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/Drawing.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Framing"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Framing.hs"]),(TargetModule (ModuleName "Hickory.ModelLoading.Bvh"),fromList ["/Users/user/dev/Hickory/core/Hickory/ModelLoading/Bvh.hs"]),(TargetModule (ModuleName "Hickory.Utils.Utils"),fromList ["/Users/user/dev/Hickory/core/Hickory/Utils/Utils.hs"]),(TargetModule (ModuleName "Hickory.ModelLoading.Parsing"),fromList ["/Users/user/dev/Hickory/core/Hickory/ModelLoading/Parsing.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Textures"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Textures.hs"]),(TargetModule (ModuleName "Hickory.Graphics.DeferredRendering"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/DeferredRendering.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Monad"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Monad.hs"]),(TargetModule (ModuleName "Hickory.Math.Interpolate"),fromList ["/Users/user/dev/Hickory/core/Hickory/Math/Interpolate.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.StockMesh"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/StockMesh.hs"]),(TargetModule (ModuleName "Hickory.Platform"),fromList ["/Users/user/dev/Hickory/core/Hickory/Platform.hs"]),(TargetModule (ModuleName "Hickory.Text.Text"),fromList ["/Users/user/dev/Hickory/core/Hickory/Text/Text.hs"]),(TargetModule (ModuleName "Hickory.Color"),fromList ["/Users/user/dev/Hickory/core/Hickory/Color.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Forward.DrawingPrimitives"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Forward/DrawingPrimitives.hs"]),(TargetModule (ModuleName "Hickory.ModelLoading.Packed"),fromList ["/Users/user/dev/Hickory/core/Hickory/ModelLoading/Packed.hs"]),(TargetModule (ModuleName "Hickory.Graphics.Shader"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/Shader.hs"]),(TargetModule (ModuleName "Hickory.Graphics.StockShaders"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/StockShaders.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Forward.ShadowPass"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Forward/ShadowPass.hs"]),(TargetFile NormalizedFilePath "/Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs",fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.RenderPass"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/RenderPass.hs"]),(TargetModule (ModuleName "Hickory.ModelLoading.Wavefront"),fromList ["/Users/user/dev/Hickory/core/Hickory/ModelLoading/Wavefront.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Text"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Text.hs"]),(TargetModule (ModuleName "Hickory.Math.Bezier"),fromList ["/Users/user/dev/Hickory/core/Hickory/Math/Bezier.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.DescriptorSet"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/DescriptorSet.hs"]),(TargetModule (ModuleName "Hickory.Text.ParseJson"),fromList ["/Users/user/dev/Hickory/core/Hickory/Text/ParseJson.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.PostProcessing"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/PostProcessing.hs"]),(TargetModule (ModuleName "Hickory.Input"),fromList ["/Users/user/dev/Hickory/core/Hickory/Input.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Vulkan"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Vulkan.hs"]),(TargetModule (ModuleName "Hickory.ModelLoading.DirectXModel"),fromList ["/Users/user/dev/Hickory/core/Hickory/ModelLoading/DirectXModel.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Utils"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Utils.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Forward.Types"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Forward/Types.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.StockTexture"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/StockTexture.hs"]),(TargetModule (ModuleName "Hickory.Camera"),fromList ["/Users/user/dev/Hickory/core/Hickory/Camera.hs"]),(TargetModule (ModuleName "Hickory.Graphics.VAO"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/VAO.hs"]),(TargetModule (ModuleName "Hickory.Graphics.ShaderMonad"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/ShaderMonad.hs"]),(TargetModule (ModuleName "Hickory.Graphics.Textures"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/Textures.hs"]),(TargetModule (ModuleName "Hickory.Utils.Projection"),fromList ["/Users/user/dev/Hickory/core/Hickory/Utils/Projection.hs"]),(TargetModule (ModuleName "Hickory.Types"),fromList ["/Users/user/dev/Hickory/core/Hickory/Types.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Forward.Renderer"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Forward/Renderer.hs"]),(TargetModule (ModuleName "Hickory.Graphics.MatrixMonad"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/MatrixMonad.hs"]),(TargetModule (ModuleName "Hickory.Graphics.DrawText"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/DrawText.hs"]),(TargetModule (ModuleName "Hickory.Text.Types"),fromList ["/Users/user/dev/Hickory/core/Hickory/Text/Types.hs"]),(TargetModule (ModuleName "Hickory.Math.Vector"),fromList ["/Users/user/dev/Hickory/core/Hickory/Math/Vector.hs"]),(TargetModule (ModuleName "Hickory.Resources"),fromList ["/Users/user/dev/Hickory/core/Hickory/Resources.hs"]),(TargetModule (ModuleName "Hickory.Graphics.SSAO"),fromList ["/Users/user/dev/Hickory/core/Hickory/Graphics/SSAO.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Mesh"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.RenderTarget"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/RenderTarget.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Forward.Lit"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Forward/Lit.hs"]),(TargetModule (ModuleName "Paths_Hickory"),fromList ["/Users/user/dev/Hickory/core/.stack-work/dist/aarch64-osx/Cabal-3.8.1.0/build/autogen/Paths_Hickory.hs"]),(TargetModule (ModuleName "Hickory.Math.Matrix"),fromList ["/Users/user/dev/Hickory/core/Hickory/Math/Matrix.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.Forward.ObjectPicking"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/Forward/ObjectPicking.hs"]),(TargetModule (ModuleName "Hickory.Vulkan.DynamicMesh"),fromList ["/Users/user/dev/Hickory/core/Hickory/Vulkan/DynamicMesh.hs"]),(TargetModule (ModuleName "Hickory.Text"),fromList ["/Users/user/dev/Hickory/core/Hickory/Text.hs"])]
2023-10-11T19:34:35.920934Z | Debug | Finished build session
AsyncCancelled
2023-10-11T19:34:35.920993Z | Debug | Restarting build session due to new component
Action Queue: [User TypeCheck]
Keys: [GhcSessionIO; , GetKnownTargets; ]
Aborting previous build session took 0.00s 
2023-10-11T19:34:35.921194Z | Debug | hlint: Getting hlint ideas for  NormalizedFilePath "/Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs"
2023-10-11T19:34:36.051744Z | Debug | hlint: Using extensions for  NormalizedFilePath "/Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs": [ MonomorphismRestriction
                                                                                                       , RelaxedPolyRec
                                                                                                       , ForeignFunctionInterface
                                                                                                       , ImplicitPrelude
                                                                                                       , ScopedTypeVariables
                                                                                                       , BangPatterns
                                                                                                       , OverloadedStrings
                                                                                                       , OverloadedLists
                                                                                                       , DisambiguateRecordFields
                                                                                                       , RecordWildCards
                                                                                                       , NamedFieldPuns
                                                                                                       , ViewPatterns
                                                                                                       , GADTSyntax
                                                                                                       , DoAndIfThenElse
                                                                                                       , BlockArguments
                                                                                                       , ConstraintKinds
                                                                                                       , PolyKinds
                                                                                                       , DataKinds
                                                                                                       , InstanceSigs
                                                                                                       , StandaloneDeriving
                                                                                                       , DeriveDataTypeable
                                                                                                       , DeriveFunctor
                                                                                                       , DeriveTraversable
                                                                                                       , DeriveFoldable
                                                                                                       , DeriveGeneric
                                                                                                       , DeriveLift
                                                                                                       , TypeSynonymInstances
                                                                                                       , FlexibleContexts
                                                                                                       , FlexibleInstances
                                                                                                       , ConstrainedClassMethods
                                                                                                       , MultiParamTypeClasses
                                                                                                       , ExistentialQuantification
                                                                                                       , EmptyDataDecls
                                                                                                       , KindSignatures
                                                                                                       , GeneralizedNewtypeDeriving
                                                                                                       , PostfixOperators
                                                                                                       , TupleSections
                                                                                                       , PatternGuards
                                                                                                       , RankNTypes
                                                                                                       , TypeOperators
                                                                                                       , ExplicitForAll
                                                                                                       , TraditionalRecordSyntax
                                                                                                       , LambdaCase
                                                                                                       , BinaryLiterals
                                                                                                       , HexFloatLiterals
                                                                                                       , DuplicateRecordFields
                                                                                                       , EmptyCase
                                                                                                       , PatternSynonyms
                                                                                                       , NamedWildCards
                                                                                                       , TypeApplications
                                                                                                       , EmptyDataDeriving
                                                                                                       , NumericUnderscores
                                                                                                       , StarIsType
                                                                                                       , ImportQualifiedPost
                                                                                                       , StandaloneKindSignatures
                                                                                                       , FieldSelectors ]

2023-10-11T19:34:36.511516Z | Debug | SUCCEEDED LOADING HIE FILE FOR /Users/user/.cache/ghcide/main-7ee3613dc5682aa8cbb222c7df23529ea959882c-7ee3613dc5682aa8cbb222c7df23529ea959882c/Hickory/Vulkan/Types.hie
2023-10-11T19:34:36.511599Z | Debug | Re-indexing hie file for /Users/user/dev/Hickory/core/Hickory/Vulkan/Types.hs
2023-10-11T19:34:36.547525Z | Debug | SUCCEEDED LOADING HIE FILE FOR /Users/user/.cache/ghcide/main-7ee3613dc5682aa8cbb222c7df23529ea959882c-7ee3613dc5682aa8cbb222c7df23529ea959882c/Hickory/Vulkan/Vulkan.hie
2023-10-11T19:34:36.547612Z | Debug | Re-indexing hie file for /Users/user/dev/Hickory/core/Hickory/Vulkan/Vulkan.hs
2023-10-11T19:34:36.664021Z | Info | updateFileDiagnostics published different from new diagnostics - file diagnostics: File:     /Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs
Hidden:   no
Range:    37:1-37:26
Source:   typecheck
Severity: DiagnosticSeverity_Warning
Message: 
  [0;93mOrphan instance: instance Binary Attribute
  Suggested fix:
  Move the instance declaration to the module of the class or of the type, or
  wrap the type with a newtype and declare the instance on the new type.[0m
File:     /Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs
Hidden:   no
Range:    38:1-38:21
Source:   typecheck
Severity: DiagnosticSeverity_Warning
Message: 
  [0;93mOrphan instance: instance Binary Mesh
  Suggested fix:
  Move the instance declaration to the module of the class or of the type, or
  wrap the type with a newtype and declare the instance on the new type.[0m
File:     /Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs
Hidden:   no
Range:    98:33-98:35
Source:   typecheck
Severity: DiagnosticSeverity_Warning
Message:  [0;93mNo variables bound in the record wildcard match are usedPossible fix: omit the ‘..’[0m
File:     /Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs
Hidden:   no
Range:    167:27-167:44
Source:   typecheck
Severity: DiagnosticSeverity_Warning
Message: 
  [0;93mThe record update zero
  {size = bufferSize} with type BufferCopy is ambiguous.
  This will not be supported by -XDuplicateRecordFields in future releases of GHC.[0m
File:     /Users/user/dev/Hickory/core/Hickory/Vulkan/Mesh.hs
Hidden:   no
Range:    185:26-185:74
Source:   typecheck
Severity: DiagnosticSeverity_Warning
Message: 
  [0;93mThe record update zero
  {flags = COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT} with type CommandBufferBeginInfo is ambiguous.
  This will not be supported by -XDuplicateRecordFields in future releases of GHC.[0m
2023-10-11T19:34:36.664821Z | Debug | Finished: User TypeCheck Took: 0.74s
2023-10-11T19:34:36.707150Z | Debug | Finished: GetHie Took: 0.02s

C2023-10-11T19:34:36.949161Z | Debug | Finished: GenerateCore Took: 0.24s
ompleted (1 file worked, 0 files failed)
```

**Comment by asivitz** [comment]
I realized (thanks to fendor on IRC) that the specific file I was testing had `{-# OPTIONS_GHC -Wno-deferred-out-of-scope-variables #-}`, and the specific error I was testing was an out of scope variable error. So this is almost entirely user error here. Although I believe I did originally add that option to suppress an HLS lint that I didn't find helpful. So perhaps there's some improvement to HLS that could be made there. But this is basically all my fault. Closing.
